docs/via_guides/data-flow.md

```diff
@@ -0,0 +1,872 @@
+sealed batches asynchronously and prepares them for the next stage in the data flow.
+
+**Database Interaction**: The commitment generator doesn't receive direct tasks from state keeper. Instead, it polls the
+database for sealed batches that need commitments using queries that look for specific conditions. This indirect
+communication pattern is central to the system's architecture, allowing components to operate independently while
+maintaining a coherent workflow.
+
+#### Creating Tasks for Downstream Components
+
+Once a batch is sealed and commitments are generated, the state keeper indirectly creates tasks for:
+
+1. [`via_da_dispatcher`](../../core/node/via_da_dispatcher): The data availability dispatcher component that:
+   - Prepares batch data for publication to the data availability layer
+   - Manages the flow of data to the Bitcoin network
+   - Handles batch metadata for proof generation
+
+#### Difference between via_state_keeper and state_keeper
+
+The [`via_state_keeper`](../../core/node/via_state_keeper) is an adaptation of the original `state_keeper` from zkSync
+Era, modified to be compatible with Bitcoin as the data availability layer. Key differences include:
+
+- Integration with Bitcoin-specific components like `via_btc_watch` instead of Ethereum components
+- Creation of tasks for `via_da_dispatcher` instead of `eth_sender`
+- Modified commitment generation process optimized for Bitcoin's data constraints
+- Adaptation to Bitcoin's transaction model for priority transactions
+
+---
+
+### Committed by sequencer/proposer on Bitcoin
+
+After a batch is sealed and its commitments are generated, the next step is to commit this data to the Bitcoin
+blockchain for data availability. This process is handled through the coordination of multiple components, primarily
+`via_da_dispatcher` and `via_btc_sender`.
+
+#### Data Availability Dispatcher
+
+The [`via_da_dispatcher`](../../core/node/via_da_dispatcher) component is responsible for preparing batch data for
+publication to the data availability layer (Bitcoin). As explained in its
+[README](../../core/node/via_da_dispatcher/README.md), it extends the functionality of the original zkSync da_dispatcher
+but with specific adaptations for Bitcoin:
+
+- It retrieves sealed L1 batches that are ready for dispatch from the database
+- It processes batch commitments and prepares them for publication
+- It handles both L1 batch commitments and L1 batch proofs for the Bitcoin data availability layer
+
+The main execution loop in [`da_dispatcher.rs`](../../core/node/via_da_dispatcher/src/da_dispatcher.rs) performs three
+concurrent tasks:
+
+1. `dispatch()` - Prepares and dispatches batch data to the data availability layer
+2. `poll_for_inclusion()` - Checks if dispatched data has been successfully included
+3. `dispatch_proofs()` - Handles proof dispatching for verified batches
```

Comment:
When we are dispatching the batch proof, the proof is not yet verified, the proof verification is what is happening next. Maybe better to say "for proven batches"